monai/transforms/post/dictionary.py

```diff
@@ -623,38 +624,53 @@ def __call__(self, data: Mapping[Hashable, Any]) -> Dict[Hashable, Any]:
             self.device,
             self.post_func,
         ):
-            transform_key = InvertibleTransform.trace_key(orig_key)
-            if transform_key not in d:
-                warnings.warn(f"transform info of `{orig_key}` is not available or no InvertibleTransform applied.")
-                continue
-
-            transform_info = d[transform_key]
+            if isinstance(d[key], MetaTensor):
+                if orig_key not in d:
+                    warnings.warn(f"transform info of `{orig_key}` is not available in MetaTensor {key}.")
+                    continue
+            else:
+                transform_key = InvertibleTransform.trace_key(orig_key)
+                if transform_key not in d:
+                    warnings.warn(f"transform info of `{orig_key}` is not available or no InvertibleTransform applied.")
+                    continue
+
+            if orig_key in d and isinstance(d[orig_key], MetaTensor):
+                transform_info = d[orig_key].applied_operations
+                meta_info = d[orig_key].meta
+            else:
+                transform_info = d[InvertibleTransform.trace_key(orig_key)]
```

Comment:
Do we still need to support this logic? I think most of the dict transforms in this PR only track `transform_info` based on the MetaTensor of array transforms now.

Thanks.

Reply:
we still have `FromMetaTensor` transforms to make use of meta dictionaries, so I think we can keep these a few lines for now for compatibility, and deprecate them in v1.0. 

btw the argument `orig_keys` could be optional now https://github.com/Project-MONAI/MONAI/pull/4562/commits/bf5fb90ce8e24692c9eab8e99d8b37508baa8bc4